ftw_tools/cli.py

```diff
@@ -536,12 +553,351 @@ def inference_run(
         gpu,
         patch_size,
         batch_size,
+        num_workers,
         padding,
         overwrite,
         mps_mode,
     )
 
 
+@inference.command(
+    "run-instance-segmentation",
+    help="Run an instance segmentation model inference on a single Sentinel-2 L2A satellite images specified via INPUT.",
+)
+@click.argument("input", type=click.Path(exists=True), required=True)
+@click.option(
+    "--model",
+    "-m",
+    type=click.Choice(["DelineateAnything", "DelineateAnything-S"]),
+    required=True,
+    help="The model to use for inference.",
+)
+@click.option(
+    "--out",
+    "-o",
+    type=str,
+    default=None,
+    help="Output filename for the polygonized data. Defaults to the name of the input file with parquet extension. "
+    + SUPPORTED_POLY_FORMATS_TXT,
+)
+@click.option(
+    "--gpu",
+    type=int,
+    help="GPU ID to use. If not provided, CPU will be used by default.",
+)
+@click.option(
+    "--image_size",
+    type=int,
+    default=320,
+    show_default=True,
+    help="Image size to use for inference.",
+)
+@click.option(
+    "--patch_size",
+    type=int,
+    default=None,
+    help="Size of patch to use for inference. Defaults to 1024 unless the image is < 1024x1024px.",
+)
```

Comment:
```suggestion
@click.option(
    "--patch_size",
    "-ps",
    type=click.IntRange(min=128),
    default=None,
    help="Size of patch to use for inference. Defaults to 1024 unless the image is < 1024x1024px and a smaller value otherwise.",
)
```

Maybe check whether that's actually what you are implementing for <1024x1024...